lib/IR/GraphScheduler.cpp

```diff
@@ -140,6 +140,26 @@ class ChildMemSizeBasedScheduler : public Scheduler {
       orderedChildren.push_back(N->getPredicate());
     }
 
+    // SaveNode hack:
+    // We don't model memory dependencies, but we still need to honor them.
+    // Make sure the SaveNode happens after the last use of the output variable.
+    if (SaveNode *save = llvm::dyn_cast<SaveNode>(N)) {
+      Variable *output = save->getVariable();
+      for (NodeUse &use : output->getUsers()) {
```

Comment:
`const auto &`?

Reply:
We cannot go with const here because we push non-const node to orderedChildren.
As for auto, I am not a fan of using it unless the type name is already nearby (like with the dyn_cast you've mentioned). To me getUsers does not obviously refers to NodeUse (espicially because getUser after that is Node :P).